File: backend/src/Exceptions/PricingUnavailableException.php
<?php

declare(strict_types=1);

namespace Quantis\AIPortfolioAssistant\Exceptions;

/**
 * Thrown when a token price cannot be resolved from system_llm_settings
 * (no row, a NULL price, or a failed DB query). Never swallowed into a
 * hardcoded fallback — callers on the billing path surface it.
 */
class PricingUnavailableException extends \RuntimeException
{
}

File: backend_python/app/exceptions/pricing_unavailable_exception.py
"""Exception when a token price cannot be resolved from system_llm_settings."""


class PricingUnavailableException(RuntimeError):
    """
    Thrown when a token price cannot be resolved from system_llm_settings
    (no row, a NULL price, or a failed DB query). Never swallowed into a
    hardcoded fallback — callers on the billing path surface it.
    """

    pass
